tests-integration/json-schema/test/json-schema.test.js

```diff
@@ -0,0 +1,169 @@
+    })
+
+    test('4. Test for Month type', () => {
+            expect(validateBasicAndAdvancedType("AdvancedTypes.StartMonth", {month: "78.9"})).toBe(true)
+    })
+})
+
+describe('Test Suite for Optional Types', () => {
+    test('Test for MayBe String', () => {
+        const optionalSchema = jsonObject["$defs"]["OptionalTypes.Assignment"]
+        expect(validator(optionalSchema, "Foo")).toBe(true)
+    })
+})
+
+describe('Test Suite for Collection Types', () => {
+    test('Test for List type', () => {
+        const listSchema = jsonObject["$defs"]["CollectionTypes.Department"]
+        expect(validator(listSchema, ["Foo", "Bam", "Baz"])).toBe(true)
+    })
+
+    test('Test for Set type', () => {
+        const setSchema = jsonObject["$defs"]["CollectionTypes.Queries"]
+        expect(validator(setSchema, ["Foo", "Bam"])).toBe(true)
+    })
+
+    test.skip('Test for Dict type', () => {
+    })
+})
+
+describe('Test Suite for Result Types', () => {
+    test('Test for basic Result type', () => {
+        const resultSchema = jsonObject["$defs"]["ResultTypes.Output"]
+        expect(validator(resultSchema, ["Ok", 56])).toBe(true)
+    })
+})
+
+describe('Test Suite for Composite Types - Records/Tuples', () => {
+
+    test('Test for Record type', () => {
+        const recordSchema = jsonObject["$defs"]["RecordTypes.Address"]
+        const recordInstance = {
+            domainName : "Foo",
+            path : "Bar",
+            protocol : "http"
+        }
+        expect(validator(recordSchema, recordInstance)).toBe(true)
+    })
+
+    test('Test for Tuple type', () => {
+        const tupleSchema = jsonObject["$defs"]["TupleTypes.Cordinates"]
+        expect(validator(tupleSchema, [34.8, 56.59, 67.5])).toBe(true)
+    })
+})
+
+describe('Test Suite for Composite Types - Custom Types', () => {
+    test('Test for Enum Type', () => {
+        const enumSchema = jsonObject["$defs"]["CustomTypes.Currencies"]
+        expect(validator(enumSchema, "EUR")).toBe(true)
+    })
+
+    test('Test for Custom type 1',  () => {
+        const custom1Schema = jsonObject["$defs"]["CustomTypes.Person"]
+        expect(validator(custom1Schema, ["Child", "Bar", 11])).toBe(true)
+    })
+
+    test('Test for Custom type 2',  () => {
+        const custom2Schema = jsonObject["$defs"]["CustomTypes.Person"]
+        expect(validator(custom2Schema, ["Adult", "Bar"])).toBe(true)
+    })
+})
+
+const validator = (schema, instance) => {
+    const ajv = new ajv2020()
+    const validate = ajv.compile(schema)
+    return validate( instance);
+}
+
+//For validation of Basic  and Advanced Types  References the main schema which is JsonObject
+const validateBasicAndAdvancedType = (typeName, instance) => {
+    const ajv = new ajv2020()
+    addFormats(ajv)
+    const schema =  {
+        "type": "object",
+        "properties": {
+            "string": {
```

Comment:
Here, you hard-coded the dummy field to be called "string", but in most of the tests you are using a different field name in the sample instance. How does this work?

Reply:
Sure. I've changed it to just "fieldName" which actually can be anything. I don't know how it works exactly, it seems ajv internally combines both the parent schema and the child schema before validation